src/database/DataTypes/ItemBatch.js

```diff
@@ -138,7 +138,7 @@ export class ItemBatch extends Realm.Object {
    * @return {Bool} Indicator whether the new vvm status should be applied to this batch.
    */
   shouldApplyVvmStatus(newVvmStatus = {}) {
-    return newVvmStatus?.id === this.currentVvmStatus?.id;
+    return newVvmStatus?.id !== this.currentVvmStatus?.id;
```

Comment:
I think this will be truthy if `newVvmStatus.id === undefined` and `currentVvmStatus.id` is defined? Intended behaviour?

Reply:
Yup - intended. Open to different ideas, but basically, if they're not the same then applying it should be OK